MoltenVK/MoltenVK/GPUObjects/MVKSwapchain.mm

```diff
@@ -615,9 +615,14 @@ static CALayerContentsGravity getCALayerContentsGravity(VkSwapchainPresentScalin
 	if (mtlLayer) {
 		NSString* screenName = @"Main Screen";
 #if MVK_MACOS && !MVK_MACCAT
-		auto* screen = mtlLayer.screenMVK;
-		if ([screen respondsToSelector:@selector(localizedName)]) {
-			screenName = screen.localizedName;
+		// To prevent deadlocks, avoid dispatching to the main thread at the cost of a less informative log.
+		if (NSThread.currentThread.isMainThread) {
+			auto* screen = mtlLayer.screenMVK;
+			if ([screen respondsToSelector:@selector(localizedName)]) {
+				screenName = screen.localizedName;
+			}
+		} else {
+			screenName = @"?";
 		}
```

Comment:
```suggestion
		// To prevent deadlocks, avoid dispatching screenMVK to the main thread at the cost of a less informative log.
		if (NSThread.isMainThread) {
			auto* screen = mtlLayer.screenMVK;
			if ([screen respondsToSelector:@selector(localizedName)]) {
				screenName = screen.localizedName;
			}
		}
```

Just cleaning this up a bit. The internal dangling `if()` will fall back to @"Main Screen". Might as well be consistent and always fall back to that.